--- ../src-base/minecraft/net/minecraft/server/MinecraftServer.java
+++ ../src-work/minecraft/net/minecraft/server/MinecraftServer.java
@@ -9,6 +9,7 @@
 import com.mojang.authlib.GameProfileRepository;
 import com.mojang.authlib.minecraft.MinecraftSessionService;
 import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
+import cn.topologycraft.uhc.GameManager;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufOutputStream;
 import io.netty.buffer.Unpooled;
@@ -141,6 +142,9 @@
     private Thread field_175590_aa;
     private long field_175591_ab = func_130071_aq();
 
+    // TC Plugin
+    protected GameManager uhcGameManager;
+
     public MinecraftServer(File p_i47054_1_, Proxy p_i47054_2_, DataFixer p_i47054_3_, YggdrasilAuthenticationService p_i47054_4_, MinecraftSessionService p_i47054_5_, GameProfileRepository p_i47054_6_, PlayerProfileCache p_i47054_7_)
     {
         this.field_110456_c = p_i47054_2_;
@@ -450,9 +454,13 @@
                 this.field_175591_ab = func_130071_aq();
                 long i = 0L;
                 this.field_147147_p.func_151315_a(new TextComponentString(this.field_71286_C));
-                this.field_147147_p.func_151321_a(new ServerStatusResponse.Version("1.12", 335));
+                // TC Plugin:
+                this.field_147147_p.func_151321_a(new ServerStatusResponse.Version("1.12-pre7", 334));
                 this.func_184107_a(this.field_147147_p);
 
+                // TC Plugin: Server Initialized Hook
+                GameManager.instance.onServerInited();
+
                 while (this.field_71317_u)
                 {
                     long k = func_130071_aq();
@@ -603,6 +611,9 @@
         this.field_71304_b.func_76320_a("root");
         this.func_71190_q();
 
+        // TC Plugin: Game Manager Update Hook
+        this.uhcGameManager.tick();
+
         if (i - this.field_147142_T >= 5000000000L)
         {
             this.field_147142_T = i;
@@ -892,7 +903,8 @@
 
     public String func_71249_w()
     {
-        return "1.12";
+        // TC Plugin:
+        return "1.12-pre7";
     }
 
     public int func_71233_x()
