--- ../src-base/minecraft/net/minecraft/world/chunk/Chunk.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/Chunk.java
@@ -3,6 +3,7 @@
 import com.google.common.base.Predicate;
 import com.google.common.collect.Maps;
 import com.google.common.collect.Queues;
+import cn.topologycraft.uhc.GameManager;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
@@ -66,6 +67,9 @@
     private final ConcurrentLinkedQueue<BlockPos> field_177447_w;
     public boolean field_189550_d;
 
+    // TC Plugin: Light Update Timer
+    private int chunkLoaded;
+
     public Chunk(World p_i1995_1_, int p_i1995_2_, int p_i1995_3_)
     {
         this.field_76652_q = new ExtendedBlockStorage[16];
@@ -1011,6 +1015,12 @@
 
     public void func_150804_b(boolean p_150804_1_)
     {
+        // TC Plugin
+        if (!GameManager.instance.isPregenerating() && chunkLoaded < 1000) {
+            this.field_76650_s = false;
+            this.field_150814_l = true;
+        }
+
         if (this.field_76650_s && this.field_76637_e.field_73011_w.func_191066_m() && !p_150804_1_)
         {
             this.func_150803_c(this.field_76637_e.field_72995_K);
@@ -1018,6 +1028,9 @@
 
         this.field_150815_m = true;
 
+        // TC Plugin
+        this.chunkLoaded++;
+
         if (!this.field_150814_l && this.field_76646_k)
         {
             this.func_150809_p();
