--- ../src-base/minecraft/net/minecraft/server/dedicated/DedicatedServer.java
+++ ../src-work/minecraft/net/minecraft/server/dedicated/DedicatedServer.java
@@ -4,6 +4,7 @@
 import com.mojang.authlib.GameProfileRepository;
 import com.mojang.authlib.minecraft.MinecraftSessionService;
 import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
+import cn.topologycraft.uhc.GameManager;
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.IOException;
@@ -107,7 +108,8 @@
         };
         thread.setDaemon(true);
         thread.start();
-        field_155771_h.info("Starting minecraft server version 1.12");
+        // TC Plugin:
+        field_155771_h.info("Starting minecraft server version 1.12-pre7");
 
         if (Runtime.getRuntime().maxMemory() / 1024L / 1024L < 512L)
         {
@@ -214,6 +216,9 @@
                     this.func_71261_m(this.field_71340_o.func_73671_a("level-name", "world"));
                 }
 
+                // TC Plugin: Save Folder Hook
+                GameManager.tryUpdateSaveFolder(this.func_71270_I());
+
                 String s = this.field_71340_o.func_73671_a("level-seed", "");
                 String s1 = this.field_71340_o.func_73671_a("level-type", "DEFAULT");
                 String s2 = this.field_71340_o.func_73671_a("generator-settings", "");
@@ -251,6 +256,10 @@
                 this.func_71191_d((this.func_71207_Z() + 8) / 16 * 16);
                 this.func_71191_d(MathHelper.func_76125_a(this.func_71207_Z(), 64, 256));
                 this.field_71340_o.func_73667_a("max-build-height", Integer.valueOf(this.func_71207_Z()));
+
+                // TC Plugin: Server preinit Hook
+                uhcGameManager = new GameManager(this, this.func_184103_al());
+
                 TileEntitySkull.func_184293_a(this.func_152358_ax());
                 TileEntitySkull.func_184294_a(this.func_147130_as());
                 PlayerProfileCache.func_187320_a(this.func_71266_T());
@@ -260,6 +269,9 @@
                 String s3 = String.format("%.3fs", (double)i1 / 1.0E9D);
                 field_155771_h.info("Done ({})! For help, type \"help\" or \"?\"", (Object)s3);
 
+                // TC Plugin: Server postinit Hook
+                uhcGameManager.postInit();
+
                 if (this.field_71340_o.func_187239_a("announce-player-achievements"))
                 {
                     this.field_71305_c[0].func_82736_K().func_82764_b("announceAdvancements", this.field_71340_o.func_73670_a("announce-player-achievements", true) ? "true" : "false");
