--- ../src-base/minecraft/net/minecraft/entity/player/EntityPlayerMP.java
+++ ../src-work/minecraft/net/minecraft/entity/player/EntityPlayerMP.java
@@ -2,6 +2,8 @@
 
 import com.google.common.collect.Lists;
 import com.mojang.authlib.GameProfile;
+import cn.topologycraft.uhc.GameManager;
+import cn.topologycraft.uhc.GamePlayer.EnumStat;
 import io.netty.buffer.Unpooled;
 import java.util.Collection;
 import java.util.Iterator;
@@ -500,7 +502,8 @@
         if (!this.field_70170_p.func_82736_K().func_82766_b("keepInventory") && !this.func_175149_v())
         {
             this.func_190776_cN();
-            this.field_71071_by.func_70436_m();
+            // TC Plugin: dropAllItemsWithoutClear
+            this.field_71071_by.dropAllItemsWithoutClear();
         }
 
         for (ScoreObjective scoreobjective : this.field_70170_p.func_96441_U().func_96520_a(IScoreCriteria.field_96642_c))
@@ -523,11 +526,21 @@
             entitylivingbase.func_191956_a(this, this.field_70744_aE, p_70645_1_);
         }
 
+        // TC Plugin
+        Entity sourceEntity = p_70645_1_.func_76364_f();
+        if (sourceEntity == null || !(sourceEntity instanceof EntityPlayerMP)) sourceEntity = p_70645_1_.func_76346_g();
+        if (sourceEntity != null && sourceEntity instanceof EntityPlayerMP) {
+            GameManager.instance.getPlayerManager().getGamePlayer((EntityPlayerMP) sourceEntity).getStat().addStat(EnumStat.PLAYER_KILLED, 1);
+        }
+
         this.func_71029_a(StatList.field_188069_A);
         this.func_175145_a(StatList.field_188098_h);
         this.func_70066_B();
         this.func_70052_a(0, false);
         this.func_110142_aN().func_94549_h();
+
+        // TC Plugin: Player Death Hook
+        GameManager.instance.onPlayerDeath(this, p_70645_1_);
     }
 
     public void func_191956_a(Entity p_191956_1_, int p_191956_2_, DamageSource p_191956_3_)
@@ -629,7 +642,23 @@
                     }
                 }
 
-                return super.func_70097_a(p_70097_1_, p_70097_2_);
+
+                // TC Plugin
+                p_70097_2_ = GameManager.instance.modifyPlayerDamage(p_70097_2_);
+
+                Entity sourceEntity = p_70097_1_.func_76364_f();
+                if (sourceEntity == null || !(sourceEntity instanceof EntityPlayerMP)) sourceEntity = p_70097_1_.func_76346_g();
+                if (sourceEntity != null && sourceEntity instanceof EntityPlayerMP) {
+                    GameManager.instance.getPlayerManager().getGamePlayer(this).getStat().addStat(EnumStat.DAMAGE_TAKEN, p_70097_2_);
+                    GameManager.instance.getPlayerManager().getGamePlayer((EntityPlayerMP) sourceEntity).getStat().addStat(EnumStat.DAMAGE_DEALT, p_70097_2_);
+                }
+
+                // TC Plugin: Player Damage Hook
+                if (super.func_70097_a(p_70097_1_, p_70097_2_)) {
+                    GameManager.instance.onPlayerDamaged(this, p_70097_1_, p_70097_2_);
+                    return true;
+                } else return false;
+                // TC Plugin ends
             }
         }
     }
@@ -1359,18 +1388,28 @@
         return (Entity)(this.field_175401_bS == null ? this : this.field_175401_bS);
     }
 
-    public void func_175399_e(Entity p_175399_1_)
+    // TC Plugin: added check param
+    public void setSpectatingEntity(Entity entityToSpectate, boolean check)
     {
         Entity entity = this.func_175398_C();
-        this.field_175401_bS = (Entity)(p_175399_1_ == null ? this : p_175399_1_);
+        this.field_175401_bS = (Entity)(entityToSpectate == null ? this : entityToSpectate);
 
-        if (entity != this.field_175401_bS)
-        {
-            this.field_71135_a.func_147359_a(new SPacketCamera(this.field_175401_bS));
-            this.func_70634_a(this.field_175401_bS.field_70165_t, this.field_175401_bS.field_70163_u, this.field_175401_bS.field_70161_v);
-        }
+        // TC Plugin: Player Spectate Hook
+        if (check) this.field_175401_bS = GameManager.instance.onPlayerSpectate(this, field_175401_bS, entity);
+
+//		if (entity != this.spectatingEntity) {
+//			this.connection.sendPacket(new SPacketCamera(this.spectatingEntity));
+//			this.setPositionAndUpdate(this.spectatingEntity.posX, this.spectatingEntity.posY, this.spectatingEntity.posZ);
+//		}
+
+        this.field_71135_a.func_147359_a(new SPacketCamera(this.field_175401_bS));
+        this.func_70634_a(this.field_175401_bS.field_70165_t, this.field_175401_bS.field_70163_u, this.field_175401_bS.field_70161_v);
     }
 
+    public void func_175399_e(Entity p_175399_1_) {
+        setSpectatingEntity(p_175399_1_, true);
+    }
+
     protected void func_184173_H()
     {
         if (this.field_71088_bW > 0 && !this.field_184851_cj)
@@ -1439,4 +1478,12 @@
     {
         return this.field_193110_cw;
     }
+
+    // TC Plugin: Heal Hook
+    @Override
+    public void func_70691_i(float healAmount) {
+        float health = this.func_110143_aJ();
+        super.func_70691_i(healAmount);
+        GameManager.instance.getPlayerManager().getGamePlayer(this).getStat().addStat(EnumStat.HEALTH_HEALED, func_110143_aJ() - health);
+    }
 }
