--- ../src-base/minecraft/net/minecraft/entity/EntityLiving.java
+++ ../src-work/minecraft/net/minecraft/entity/EntityLiving.java
@@ -685,6 +685,12 @@
         return true;
     }
 
+    // TC Plugin adds interface for invoking
+    public void invokeDespawnEntity()
+    {
+        this.func_70623_bb();
+    }
+
     protected void func_70623_bb()
     {
         if (this.field_82179_bU)
@@ -702,12 +708,16 @@
                 double d2 = entity.field_70161_v - this.field_70161_v;
                 double d3 = d0 * d0 + d1 * d1 + d2 * d2;
 
-                if (this.func_70692_ba() && d3 > 16384.0D)
+                // TC Plugin: remove immediately despawn, replace with randomly despawn
+//                if (this.canDespawn() && d3 > 16384.0D)
+                if (this.func_70692_ba() && d3 > 16384.0D && this.field_70146_Z.nextInt(200) == 0)
                 {
                     this.func_70106_y();
                 }
 
-                if (this.field_70708_bq > 600 && this.field_70146_Z.nextInt(800) == 0 && d3 > 1024.0D && this.func_70692_ba())
+                // TC Plugin: make randomly despawn easier
+//                if (this.idleTime > 600 && this.rand.nextInt(800) == 0 && d3 > 1024.0D && this.canDespawn())
+                if (this.field_70708_bq > 300 && this.field_70146_Z.nextInt(200) == 0 && d3 > 1024.0D && this.func_70692_ba())
                 {
                     this.func_70106_y();
                 }
@@ -722,9 +732,11 @@
     protected final void func_70626_be()
     {
         ++this.field_70708_bq;
-        this.field_70170_p.field_72984_F.func_76320_a("checkDespawn");
-        this.func_70623_bb();
-        this.field_70170_p.field_72984_F.func_76319_b();
+        // TC Plugin
+        // moves the despawn logic into net.minecraft.world.World#updateEntityWithOptionalForce
+//        this.world.profiler.startSection("checkDespawn");
+//        this.despawnEntity();
+//        this.world.profiler.endSection();
         this.field_70170_p.field_72984_F.func_76320_a("sensing");
         this.field_70723_bA.func_75523_a();
         this.field_70170_p.field_72984_F.func_76319_b();
