--- ../src-base/minecraft/net/minecraft/server/management/PlayerList.java
+++ ../src-work/minecraft/net/minecraft/server/management/PlayerList.java
@@ -4,6 +4,7 @@
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import com.mojang.authlib.GameProfile;
+import cn.topologycraft.uhc.GameManager;
 import io.netty.buffer.Unpooled;
 import java.io.File;
 import java.net.SocketAddress;
@@ -341,6 +342,9 @@
             p_72377_1_.field_71135_a.func_147359_a(new SPacketPlayerListItem(SPacketPlayerListItem.Action.ADD_PLAYER, new EntityPlayerMP[] {this.field_72404_b.get(i)}));
         }
 
+        // TC Plugin: Player Join Hook
+        GameManager.instance.onPlayerJoin(p_72377_1_);
+
         worldserver.func_72838_d(p_72377_1_);
         this.func_72375_a(p_72377_1_, (WorldServer)null);
     }
@@ -527,6 +531,9 @@
             entityplayermp.func_70107_b(entityplayermp.field_70165_t, entityplayermp.field_70163_u + 1.0D, entityplayermp.field_70161_v);
         }
 
+        // TC Plugin: Player Respawn Hook
+        if (!p_72368_3_) GameManager.instance.onPlayerRespawn(entityplayermp);
+
         entityplayermp.field_71135_a.func_147359_a(new SPacketRespawn(entityplayermp.field_71093_bK, entityplayermp.field_70170_p.func_175659_aa(), entityplayermp.field_70170_p.func_72912_H().func_76067_t(), entityplayermp.field_71134_c.func_73081_b()));
         BlockPos blockpos2 = worldserver.func_175694_M();
         entityplayermp.field_71135_a.func_147364_a(entityplayermp.field_70165_t, entityplayermp.field_70163_u, entityplayermp.field_70161_v, entityplayermp.field_70177_z, entityplayermp.field_70125_A);
