--- ../src-base/minecraft/net/minecraft/world/gen/ChunkGeneratorSettings.java
+++ ../src-work/minecraft/net/minecraft/world/gen/ChunkGeneratorSettings.java
@@ -350,30 +350,30 @@
             this.field_177936_aa = 10;
             this.field_177937_ab = 0;
             this.field_177934_ac = 80;
-            this.field_177935_ad = 17;
+            this.field_177935_ad = 10;
             this.field_177941_ae = 20;
             this.field_177943_af = 0;
             this.field_177938_ag = 128;
-            this.field_177939_ah = 9;
-            this.field_177922_ai = 20;
+            this.field_177939_ah = 6;
+            this.field_177922_ai = 4;
             this.field_177924_aj = 0;
             this.field_177918_ak = 64;
-            this.field_177920_al = 9;
-            this.field_177930_am = 2;
+            this.field_177920_al = 4;
+            this.field_177930_am = 1;
             this.field_177932_an = 0;
-            this.field_177926_ao = 32;
-            this.field_177928_ap = 8;
-            this.field_177908_aq = 8;
+            this.field_177926_ao = 20;
+            this.field_177928_ap = 4;
+            this.field_177908_aq = 4;
             this.field_177906_ar = 0;
             this.field_177904_as = 16;
-            this.field_177902_at = 8;
+            this.field_177902_at = 4;
             this.field_177916_au = 1;
             this.field_177914_av = 0;
             this.field_177912_aw = 16;
-            this.field_177910_ax = 7;
+            this.field_177910_ax = 4;
             this.field_177897_ay = 1;
-            this.field_177895_az = 16;
-            this.field_177889_aA = 16;
+            this.field_177895_az = 20;
+            this.field_177889_aA = 10;
         }
 
         public boolean equals(Object p_equals_1_)
