--- ../src-base/minecraft/net/minecraft/world/gen/ChunkGeneratorOverworld.java
+++ ../src-work/minecraft/net/minecraft/world/gen/ChunkGeneratorOverworld.java
@@ -3,6 +3,9 @@
 import java.util.List;
 import java.util.Random;
 import javax.annotation.Nullable;
+import cn.topologycraft.uhc.gen.MapGenBonusChest;
+import cn.topologycraft.uhc.gen.MapGenEnderAltar;
+import cn.topologycraft.uhc.gen.MapGenMerchants;
 import net.minecraft.block.BlockFalling;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.EnumCreatureType;
@@ -403,6 +406,11 @@
             }
         }
 
+        // TC Plugin: Generates
+        new MapGenMerchants().func_180709_b(field_185995_n, field_185990_i, blockpos);
+        new MapGenBonusChest().func_180709_b(field_185995_n, field_185990_i, blockpos);
+        new MapGenEnderAltar().func_180709_b(field_185995_n, field_185990_i, blockpos);
+
         if (biome != Biomes.field_76769_d && biome != Biomes.field_76786_s && this.field_186000_s.field_177781_A && !flag && this.field_185990_i.nextInt(this.field_186000_s.field_177782_B) == 0)
         {
             int i1 = this.field_185990_i.nextInt(16) + 8;
