--- ../src-base/minecraft/net/minecraft/potion/PotionEffect.java
+++ ../src-work/minecraft/net/minecraft/potion/PotionEffect.java
@@ -2,6 +2,7 @@
 
 import com.google.common.collect.ComparisonChain;
 import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.init.MobEffects;
 import net.minecraft.nbt.NBTTagCompound;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -56,7 +57,9 @@
             field_180155_a.warn("This method should only be called for matching effects!");
         }
 
-        if (p_76452_1_.field_76461_c > this.field_76461_c)
+        // TC Plugin: Backport 1.13.2 potion effect logic, in order to fix the bug that
+        // eating gapple after eating e-gapple can reset the ABSORPTION to e-gapple level
+        if (p_76452_1_.field_76461_c > this.field_76461_c || this.field_188420_b == MobEffects.field_76444_x)
         {
             this.field_76461_c = p_76452_1_.field_76461_c;
             this.field_76460_b = p_76452_1_.field_76460_b;
