--- ../src-base/minecraft/net/minecraft/item/crafting/CraftingManager.java
+++ ../src-work/minecraft/net/minecraft/item/crafting/CraftingManager.java
@@ -1,5 +1,7 @@
 package net.minecraft.item.crafting;
 
+import cn.topologycraft.uhc.recipe.RecipeArmorRepair;
+import cn.topologycraft.uhc.recipe.RecipeGoldenApple;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.JsonObject;
@@ -53,6 +55,10 @@
             func_193379_a("banneraddpattern", new RecipesBanners.RecipeAddPattern());
             func_193379_a("shielddecoration", new ShieldRecipes.Decoration());
             func_193379_a("shulkerboxcoloring", new ShulkerBoxRecipes.ShulkerBoxColoring());
+
+            // TC Plugin
+            func_193379_a("goldenapple", new RecipeGoldenApple());
+            func_193379_a("armorrepair", new RecipeArmorRepair());
             return func_192420_c();
         }
         catch (Throwable var1)
