--- ../src-base/minecraft/net/minecraft/util/DamageSource.java
+++ ../src-work/minecraft/net/minecraft/util/DamageSource.java
@@ -50,6 +50,12 @@
         return new EntityDamageSource("mob", p_76358_0_);
     }
 
+    // TC Plugin
+    public static DamageSource causeEntityDamage(Entity entity)
+    {
+        return new EntityDamageSource("mob", entity);
+    }
+
     public static DamageSource func_188403_a(Entity p_188403_0_, EntityLivingBase p_188403_1_)
     {
         return new EntityDamageSourceIndirect("mob", p_188403_0_, p_188403_1_);
