--- ../src-base/minecraft/net/minecraft/util/DamageSource.java
+++ ../src-work/minecraft/net/minecraft/util/DamageSource.java
@@ -34,6 +34,10 @@
     public static final DamageSource field_82729_p = new DamageSource("fallingBlock");
     public static final DamageSource field_188407_q = (new DamageSource("dragonBreath")).func_76348_h();
     public static final DamageSource field_191552_t = (new DamageSource("fireworks")).func_94540_d();
+
+    // TC Plugin: Space Collapse Damage
+    public static final DamageSource spaceCollapse = new DamageSource("spaceCollapse").func_76348_h();
+
     private boolean field_76374_o;
     private boolean field_76385_p;
     private boolean field_151520_r;
@@ -50,6 +54,12 @@
         return new EntityDamageSource("mob", p_76358_0_);
     }
 
+    // TC Plugin
+    public static DamageSource causeEntityDamage(Entity entity)
+    {
+        return new EntityDamageSource("mob", entity);
+    }
+
     public static DamageSource func_188403_a(Entity p_188403_0_, EntityLivingBase p_188403_1_)
     {
         return new EntityDamageSourceIndirect("mob", p_188403_0_, p_188403_1_);
