--- ../src-base/minecraft/net/minecraft/network/NetHandlerPlayServer.java
+++ ../src-work/minecraft/net/minecraft/network/NetHandlerPlayServer.java
@@ -4,6 +4,7 @@
 import com.google.common.primitives.Doubles;
 import com.google.common.primitives.Floats;
 import com.google.common.util.concurrent.Futures;
+import cn.topologycraft.uhc.GameManager;
 import io.netty.util.concurrent.Future;
 import io.netty.util.concurrent.GenericFutureListener;
 import java.io.IOException;
@@ -773,15 +774,36 @@
 
             if (entity != null)
             {
-                this.field_147369_b.func_175399_e(this.field_147369_b);
-                this.field_147369_b.func_184210_p();
+                // TC Plugin: Spectate Adjustment
+                Entity lastSpectateTarget = field_147369_b.func_175398_C();
+                Entity origin = entity;
+                entity = GameManager.instance.onPlayerSpectate(field_147369_b, entity, lastSpectateTarget);
+                boolean observe = GameManager.instance.onPlayerSpectate(field_147369_b, field_147369_b, lastSpectateTarget) == field_147369_b;
+                // TC Plugin ends
 
                 if (entity.field_70170_p == this.field_147369_b.field_70170_p)
                 {
+                    // TC Plugin
+                    if (entity == lastSpectateTarget) return;
+                    this.field_147369_b.setSpectatingEntity(null, false);
+                    this.field_147369_b.func_184210_p();
+                    // TC Plugin ends
+
                     this.field_147369_b.func_70634_a(entity.field_70165_t, entity.field_70163_u, entity.field_70161_v);
+
+                    // TC Plugin
+                    if (!observe) {
+                        field_147369_b.setSpectatingEntity(observe ? null : entity, false);
+                    }
+                    // TC Plugin ends
                 }
                 else
                 {
+                    // TC Plugin
+                    this.field_147369_b.setSpectatingEntity(null, false);
+                    this.field_147369_b.func_184210_p();
+                    // TC Plugin ends
+
                     WorldServer worldserver1 = this.field_147369_b.func_71121_q();
                     WorldServer worldserver2 = (WorldServer)entity.field_70170_p;
                     this.field_147369_b.field_71093_bK = entity.field_71093_bK;
@@ -801,6 +823,12 @@
                     this.field_147369_b.func_70029_a(worldserver2);
                     this.field_147367_d.func_184103_al().func_72375_a(this.field_147369_b, worldserver1);
                     this.field_147369_b.func_70634_a(entity.field_70165_t, entity.field_70163_u, entity.field_70161_v);
+
+                    // TC Plugin
+                    if (!observe) {
+                        field_147369_b.setSpectatingEntity(observe ? null : entity, false);
+                    }
+
                     this.field_147369_b.field_71134_c.func_73080_a(worldserver2);
                     this.field_147367_d.func_184103_al().func_72354_b(this.field_147369_b, worldserver2);
                     this.field_147367_d.func_184103_al().func_72385_f(this.field_147369_b);
@@ -922,7 +950,8 @@
             {
                 this.func_147361_d(s);
             }
-            else
+            // TC Plugin: Player Chat Hook
+            else if (GameManager.instance.onPlayerChat(field_147369_b, s))
             {
                 ITextComponent itextcomponent = new TextComponentTranslation("chat.type.text", new Object[] {this.field_147369_b.func_145748_c_(), s});
                 this.field_147367_d.func_184103_al().func_148544_a(itextcomponent, false);
