--- ../src-base/minecraft/net/minecraft/item/ItemStack.java
+++ ../src-work/minecraft/net/minecraft/item/ItemStack.java
@@ -4,6 +4,8 @@
 import com.google.common.collect.Multimap;
 import java.text.DecimalFormat;
 import java.util.Random;
+import cn.topologycraft.uhc.GameManager;
+import cn.topologycraft.uhc.GamePlayer.EnumStat;
 import javax.annotation.Nullable;
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.block.Block;
@@ -24,6 +26,7 @@
 import net.minecraft.nbt.NBTBase;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTTagString;
 import net.minecraft.stats.StatList;
 import net.minecraft.util.ActionResult;
 import net.minecraft.util.EnumActionResult;
@@ -345,6 +348,11 @@
         if (flag)
         {
             p_179548_4_.func_71029_a(StatList.func_188057_b(this.field_151002_e));
+
+            // TC Plugin
+            if (p_179548_2_.func_177230_c() == Blocks.field_150482_ag) {
+                GameManager.instance.getPlayerManager().getGamePlayer(p_179548_4_).getStat().addStat(EnumStat.DIAMOND_FOUND, 1);
+            }
         }
     }
 
