--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketPlayerListHeaderFooter.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketPlayerListHeaderFooter.java
@@ -11,6 +11,18 @@
     private ITextComponent field_179703_a;
     private ITextComponent field_179702_b;
 
+    // TC Plugin: setter
+    public void setHeader(ITextComponent header)
+    {
+        this.field_179703_a = header;
+    }
+
+    // TC Plugin: setter
+    public void setFooter(ITextComponent footer)
+    {
+        this.field_179702_b = footer;
+    }
+
     public void func_148837_a(PacketBuffer p_148837_1_) throws IOException
     {
         this.field_179703_a = p_148837_1_.func_179258_d();
